include/picongpu/plugins/randomizedParticleMerger/VoronoiCell.hpp

```diff
@@ -0,0 +1,368 @@
+        /** necessary for probalic algorithm. expected number of particles,
+            which should be obtained after merging the particles in the current cell */
+        float_X expectedNumMacroParticles;
+        /** number of macroparticles in parent cell */
+        uint32_t parentNumMacroParticles;
+        /** necessary for probalic algorithm.  expected number of particles,
+            which should be obtained after merging the particles in the parent cell */
+        float_X parentExpectedNumMacroParticles;
+
+        HDINLINE
+        VoronoiCell(
+            VoronoiSplittingStage splittingStage = VoronoiSplittingStage::position,
+            float_X parentNumMacroParticles = 0.0_X,
+            float_X parentExpectedNumMacroParticles = float_X(-1.0)
+        ) :
+            status( VoronoiStatus::collecting ),
+            splittingStage( splittingStage ),
+            numMacroParticles( 0u ),
+            numRealParticles( float_X( 0.0_X ) ),
+            meanMomentumValue( float3_X::create( 0.0_X ) ),
+            meanPositionValue( float3_X::create( 0.0_X ) ),
+            meanMomentumSquaredValue( float3_X::create( 0.0_X ) ),
+            meanPositionSquaredValue( float3_X::create( 0.0_X ) ),
+            firstParticleFlag( 0 ),
+            expectedNumMacroParticles ( 0.0_X ),
+            parentNumMacroParticles ( parentNumMacroParticles ),
+            parentExpectedNumMacroParticles ( parentExpectedNumMacroParticles )
+
+        {}
+
+        /** status setter */
+        HDINLINE
+        void setToAbort()
+        {
+            status = VoronoiStatus::abort;
+        }
+
+
+        /** Mark the cell for splitting
+         *
+         * @param splittingComponent index of position or momentum component
+         *                           to use for splitting
+         * @param lowerCellId cell index of a new "lower" subcell
+         * @param higherCellId cell index of a new "upper" subcell
+         */
+        HDINLINE
+        void setToSplitting(
+            const uint8_t splittingComponent,
+            const int32_t lowerCellId,
+            const int32_t higherCellId)
+        {
+            status = VoronoiStatus::splitting;
+            this->splittingComponent = splittingComponent;
+            this->lowerCellId = lowerCellId;
+            this->higherCellId = higherCellId;
+        }
+
+
+        /** status setter */
+        HDINLINE
+        void setToReadyForMerging()
+        {
+            this->status = VoronoiStatus::readyForMerging;
+        }
+
+        /** check if the current thread is associated to the first particle */
+        template< typename T_Acc >
+        DINLINE
+        bool isFirstParticle(T_Acc const & acc)
+        {
+            return atomicExch( &this->firstParticleFlag, 1 ) == 0;
+        }
+
+
+        /** add a particle to this Voronoi cell */
+        template< typename T_Acc >
+        DINLINE
+        void addParticle(
+            T_Acc const & acc,
```

Comment:
we use normally right hand side const as in this line but since you used mostly left hand side keep the left hand side const and change this line to `const T_Acc` to use one stile within this file.